### PR TITLE
feature/conditionDialog

### PR DESCRIPTION
**Pull request by LegoFed3**
- Add dialog on condition drop on sheet/token to select its duration (SHIFT-skippable as usual)
- Fix rendering of condition links in power previews
- Fix ongoing damage and system macro created AEs not appearing on tokens

TODOs:
- [x] Condition duration dialog on drag
- [x] shift bypass
- [x] handle token toggle
- [x] handle *sourceTurn
- [x] fix start/end-of-turn ongoing draggables not producing an ongoing AE
- [x] store source combat round to properly support *next* turn

### CODE REVIEW

File: src/module/archmage.js [hunk 1]
```diff
@@ -1143,9 +1123,56 @@ Hooks.on('dropCanvasData', async (canvas, data) => {
...
+          duration = html.find('[name="duration"]').val();
+          let options = []
+          if (['StartOfNextSourceTurn', 'EndOfNextSourceTurn'].includes(duration)) {
+            options = {sourceTurnUuid: source};
+          }
+          game.archmage.MacroUtils.setDuration(effectData, duration, options);
+          if ( token ) return token._object.toggleEffect(effectData, {active: true});
+          else return actor.createEmbeddedDocuments("ActiveEffect", [effectData]);
+        }
+      }
```
Comment (ben) [inline src/module/archmage.js:1172]: Any reason you don't just put the contents of this `if` into the `apply` callback?

File: src/module/archmage.js [hunk 2]
```diff
@@ -1143,9 +1123,56 @@ Hooks.on('dropCanvasData', async (canvas, data) => {
...
     }
-    return token.actor.createEmbeddedDocuments("ActiveEffect", [effectData]);
+    return await _applyAEDurationDialog(actor, effectData, data.ends, data.source, token);
   }
-});
+}
+
+async function _applyAEDurationDialog(actor, effectData, duration, source, token=undefined) {
+  if (event.shiftKey) {
+    if ( token ) return token.toggleEffect(effectData, {active: true});
```
Comment (ben) [inline src/module/archmage.js:1132]: This call doesn't work in v12

Comment (ben) [reply src/module/archmage.js]: Pushed a fix, just had to add a `._object` in the chain.

File: src/module/archmage.js (hunk 1 above)
Comment (LegoFed3) [reply src/module/archmage.js]: With the current version of the code, none really. It began with a different idea and I didn't quite cleaned it up yet.

**Review by asacolips — APPROVED** [review]

**Pull request merged**